libyul/backends/evm/EVMDialect.cpp

```diff
@@ -396,6 +393,50 @@ std::vector<std::optional<BuiltinFunctionForEVM>> createBuiltins(langutil::EVMVe
 				_assembly.appendAuxDataLoadN(static_cast<uint16_t>(literal->value.value()));
 			}
 		));
+
+		builtins.emplace_back(createFunction(
+			"eofcreate",
+			5,
+			1,
+			EVMDialect::sideEffectsOfInstruction(evmasm::Instruction::EOFCREATE),
+			ControlFlowSideEffects::fromInstruction(evmasm::Instruction::EOFCREATE),
+			{LiteralKind::String, std::nullopt, std::nullopt, std::nullopt, std::nullopt},
+			[](
+				FunctionCall const& _call,
+				AbstractAssembly& _assembly,
+				BuiltinContext& context
+			) {
+				yulAssert(_call.arguments.size() == 5);
+				Literal const* literal = std::get_if<Literal>(&_call.arguments.front());
+				auto const* containerID = valueOrNullptr(context.subIDs, formatLiteral(*literal));
```

Comment:
The way we resolve suboject names here (and in `returncontract()`) must match what we're doing in `AsmAnalysis`. Currently it mostly does, but only because we can assume the name is validated (so it's not a path) and because `subIDs` contains both qualified and unqualified subobject names. The still leaves one case that is broken: object and subobject having the same name. Using unqualified name in that case is ambiguous an needs a fix.

It would be best to have some helper(s) that can be used both here and in `AsmAnalysis` so that this lookup is always consistent by design. `Object`/`ObjectNode` is probably the best place for such helpers.

Reply:
Also, you should assert here that object name does not contain dots. Maybe we should make a helper for this too (e.g.`ObjectNode::isUnqualifiedName()`).

Reply:
Now it's validated thnaks to https://github.com/ethereum/solidity/pull/15536.
Asset added.